contents/rust/Rust.md

```diff
@@ -0,0 +1,79 @@
+
+### Safe
+
+Designing the language to be safe allows common safety errors such as segmentation faults ([[1]](https://en.wikipedia.org/wiki/Segmentation_fault), [[2]](https://stackoverflow.com/questions/2346806/what-is-a-segmentation-fault/2346849#2346849)), [resource leaks](http://blog.skylight.io/rust-means-never-having-to-close-a-socket/), and [many others](https://www.reddit.com/r/rust/comments/2mwpie/what_are_the_advantages_of_rust_over_modern_c/) to be avoided. Many of these pitfalls can already be avoided in modern languages such as [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) in C++ and [try-with-resources](https://blogs.oracle.com/darcy/more-concise-try-with-resources-statements-in-jdk-9) in Java, but mostly still requires developer discipline to ensure safety ([[1]](https://www.rust-lang.org/en-US/faq.html#why-rust-vs-cxx), [[2]](https://www.reddit.com/r/rust/comments/2mwpie/what_are_the_advantages_of_rust_over_modern_c/)).
+
+The underlying concept for the safety provided in Rust is called [ownership](https://doc.rust-lang.org/book/second-edition/ch04-00-understanding-ownership.html). Intuitively, the concept just means that each value has only a single owner (i.e. a variable) and if it the value needs to be shared, it can be borrowed. This is a powerful concept that ensures safety in Rust compile-time and is [can be applied to other languages](https://codewithoutrules.com/2017/01/26/object-ownership/).
+
+### Concurrent
+
+Concurrency is getting [increasingly important](https://softwareengineering.stackexchange.com/questions/115474/why-should-i-know-concurrent-programming) but it is challenging to write concurrent code ([[1]](https://news.ycombinator.com/item?id=8138578), [[2]](https://golang.org/doc/faq#csp), [[3]](http://joeduffyblog.com/2016/11/30/15-years-of-concurrency/)).
+
+Rust provides concurrency which is built upon the safety concepts. The implication is that the safety concepts allows us to be [fearless when writing concurrent code](https://blog.rust-lang.org/2015/04/10/Fearless-Concurrency.html) by helping point out mistakes compile time. 
+
+### Practical
+
+Rust is designed to be practical, as shown in Rust's guiding design:
+* Uses old established techniques instead of particularly cutting-edge technologies
+* Provides only majority-case features
+* Runs on widely used platforms without unnecessary compromises
+
+See [non-goals](https://www.rust-lang.org/en-US/faq.html#what-are-some-non-goals).
+
+#### Production Ready
+Also, in order for a language to be practical it must be usable in production. The commonly well known large project is [Servo](https://github.com/servo/servo), the prototype browser engine Mozilla is working on, but there are also many [organizations running Rust in production](https://www.rust-lang.org/en-US/friends.html). As an example, Jamie Turner from Dropbox [explains the reasons for using Rust](https://news.ycombinator.com/item?id=11283688).
```

Comment:
`commonly well known large project` -> `well known large project for Rust`

Reply:
`the prototype browser engine Mozilla is working on, but there are also many [organizations running Rust in production]` -> `the prototype browser engine that Mozilla is working on. On top of that, there are many other [organizations running Rust in production] ...`